tabbycat/utils/tables.py

```diff
@@ -846,15 +845,13 @@ def add_debate_ballot_link_column(self, debates, show_ballot=False):
             self.add_column(ballot_links_header, ballot_links_data)
 
         elif self.private_url:
-            debates = Debate.objects.filter(pk__in=[d.pk for d in debates]).select_related('round').annotate(
-                has_ballot=Exists(BallotSubmission.objects.filter(debate_id=OuterRef('id')).exclude(discarded=True)),
-            ).prefetch_related(
-                Prefetch('ballotsubmission_set', queryset=BallotSubmission.objects.exclude(discarded=True), to_attr='nondiscard_ballots'))
+            populate_debate_has_ballots(debates)
             ballot_links_data = []
             for debate in debates:
-                if not debate.has_ballot:
+                dummy_bs = BallotSubmission(debate=debate, single_adj=self.tournament.pref('individual_ballots'))
+                if not debate._has_ballot:
                     ballot_links_data.append(_("No ballot"))
-                elif not get_result_class(debate.nondiscard_ballots[0], debate.round, self.tournament).uses_speakers:
+                elif not get_result_class(dummy_bs, debate.round, self.tournament).uses_speakers:
```

Comment:
Would it be cleaner to refactor `get_result_class` so that passing in a `BallotSubmission` isn't mandatory if there's another way to get the required information? _e.g._ permit passing in `single_adj` instead of `ballotsub` (but still generally expect `ballotsub`)? (We should probably add a docstring to `get_result_class`, I guess.)